drivers/can/can_mcux_flexcan.c

```diff
@@ -467,8 +469,12 @@ static int mcux_flexcan_send(const struct device *dev,
 	xfer.frame = &data->tx_cbs[alloc].frame;
 	xfer.mbIdx = ALLOC_IDX_TO_TXMB_IDX(alloc);
 	FLEXCAN_SetTxMbConfig(config->base, xfer.mbIdx, true);
+	k_mutex_lock(&data->tx_mutex, K_FOREVER);
+	irq_disable(config->irq_n);
```

Comment:
Did not check if there is a race condition, but a spin lock would we appropriate here instead if irq disable and a mutex.

Reply:
@alexanderwachter, my proposed solution affects only CAN interrupts and all threads working with CAN. If we use a spin lock, we will make all threads wait. The race condition can be seen for example [here](https://github.com/zephyrproject-rtos/hal_nxp/blob/master/mcux/mcux-sdk/drivers/flexcan/fsl_flexcan.h#L1676-L1679).

Reply:
IRQ disable prevents any other thread to run too. How can the thread be interrupted if not from an irq?

Reply:
Ah. Now I see. You want to only disable the send irq?

Reply:
I suggest disable all interrupts.